tests/GetHomeDirectory002.hs

```diff
@@ -0,0 +1,18 @@
+{-# LANGUAGE CPP #-}
+module GetHomeDirectory002 where
+
+#if !defined(mingw32_HOST_OS)
+import System.Posix.Env
+#endif
+#include "util.inl"
+
+-- Test that the getpwuid_r fallback works.
+-- This is only relevant on unix.
```

Comment:
Does this need to be Unix-specific?  I imagine it should succeed on Windows too.

Reply:
Windows does not have `getpwuid_r`.

Reply:
The test should still work on Windows since `getHomeDirectory` has a different implementation on Windows that does not rely on `HOME`.  Thus, there is no reason to complicate this test with `#if`s.

Reply:
That does not work. `unsetEnv` does not exist on windows.

Reply:
Oh, I see.  You are using `unsetEnv` from `unix` not `base`.  In that case, let's leave this be.